### PR TITLE
[#81] fix signal handler

### PR DESCRIPTION
**Pull request by elfenpiff**
## Notes for Reviewer

The original problem was that I defined a struct `sigaction_t` in rust posix types and reinterpreted it has the posix c type `sigaction`, this worked for all platforms, but this was pure luck since posix states nothing about the member order or hidden members. It is not possible to use the `sigaction` struct directly from bindgen since some libc implementation have weird C macro hacks. So there are posix libc implementations that strictly speaking violate the posix standard and correct this with a macro called `sa_handler` (the member they renamed to something else).
With bindgen it ends up with different struct member names on different libc implementation therefore, there was nothing else possible but to create an `iox2_sigaction` struct in `posix.h` and do the translation in the c function `iox2_sigaction_func` since C supports these weird macro hacks in a platform independent way.

## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [x] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

Use either 'Closes #123' or 'Relates to #123' to reference the corresponding issue.

Relates to #81

### CODE REVIEW

File: iceoryx2-pal/posix/build.rs [hunk 1]
```diff
@@ -20,13 +20,9 @@ fn main() {
     #[cfg(any(target_os = "linux", target_os = "freebsd"))]
     println!("cargo:rustc-link-lib=pthread");
 
-    if std::env::var("DOCS_RS").is_ok() {
-        println!("cargo:rerun-if-changed=src/c/posix_docs_rs.h");
```
Comment (elfenpiff) [inline iceoryx2-pal/posix/build.rs:24]: This was obsolete code which is overlooked. The docs rs issue is addressed with the macro `IOX2_DOCS_RS_SUPPORT` which is defined below.

File: iceoryx2-pal/posix/src/c/sigaction.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    if (oact != NULL) {
+        tr_oact_ptr = &tr_oact;
+    }
+
+    int ret_val = sigaction(sig, tr_act_ptr, tr_oact_ptr);
+
+    if (ret_val == 0 && oact != NULL) {
+        oact->iox2_sa_flags = tr_act.sa_flags;
+        oact->iox2_sa_mask = tr_act.sa_mask;
+        oact->iox2_sa_handler = (size_t)tr_act.sa_handler;
```
Comment (elBoberido) [inline iceoryx2-pal/posix/src/c/sigaction.c:42]:
```suggestion
        oact->iox2_sa_flags = tr_oact.sa_flags;
        oact->iox2_sa_mask = tr_oact.sa_mask;
        oact->iox2_sa_handler = (size_t)tr_oact.sa_handler;
```
I think you mixed up the variable

File: iceoryx2-pal/posix/src/c/socket_macros.c [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
 #ifdef _WIN64
+// clang-format off
+// the include order is important, since some headers are defining macros that
+// are used in the next header
 #include <WinSock2.h>
 #include <Windows.h>
 #include <MSWSock.h>
 #include <io.h>
```
Comment (elBoberido) [inline iceoryx2-pal/posix/src/c/socket_macros.c:20]: Oh boy. I'm already looking forward to implement MinGW support for iceoryx2 :sweat_smile:

**Review by elBoberido — COMMENTED** [review]
Looks good but I think you mixed up one variable

File: iceoryx2-pal/posix/src/c/socket_macros.c (hunk 3 above)
Comment (elfenpiff) [reply iceoryx2-pal/posix/src/c/socket_macros.c:20]: Hehe those things are a lot of fun, and they remember me of my old university job where bash script A declared a variable, that was modified in bash script B and C and D used it and when for whatever reason the scripts were called in different order weird things happened.

**Review by elBoberido — APPROVED** [review]

**Pull request merged**